types/index.d.ts

```diff
@@ -7,80 +7,188 @@
 import * as stream from 'stream';
 import { EventEmitter } from 'events'
 
-type Schema = string | object;  // TODO object should be further specified
+//"virtual" namespace (no JS, just types) for Avro Schema
+declare namespace schema
+{
+  export type AvroSchema = DefinedType | DefinedType[];
```

Comment:
Does this allow references? (References can be arbitrary strings, pointing to values in a registry.)

Reply:
ooh, perhaps not. I haven't used references in my project so far. It would be easy to fix, to add a string type to the DefinedType (along with the primitive, complex and logical types). The primitive type than becomes a kind of redundant as they are just a collection of strings, though it still helps in code-completion scenarios.